### PR TITLE
dev-db/soci: new package, dependency of rstudio

### PR DESCRIPTION
**Pull request by ezzieyguywuf**
Split from PR # 19865.

See [here][1] for some of the conversation

[1]: https://github.com/gentoo/gentoo/pull/19865#pullrequestreview-629990295

Signed-off-by: Wolfgang E. Sanyer <ezzieyguywuf@gmail.com>

### CODE REVIEW

**Comment by ezzieyguywuf** [comment]
@juippis ping

File: dev-libs/soci/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>ezzieyguywuf@gmail.com</email>
+		<name>Wolfgang E. Sanyer</name>
+	</maintainer>
+	<maintainer type="project">
```
Comment (AndrewAmmerlaan) [inline dev-libs/soci/metadata.xml:8]:
```suggestion
	<maintainer type="project" proxied="proxy">
```

File: dev-libs/soci/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
```
Comment (AndrewAmmerlaan) [inline dev-libs/soci/metadata.xml:4]:
```suggestion
	<maintainer type="person" proxied="yes">
```

File: dev-libs/soci/soci-4.0.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# Copyright 2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (AndrewAmmerlaan) [inline dev-libs/soci/soci-4.0.1.ebuild:4]: Could we make this `EAPI=8` please?

File: dev-libs/soci/soci-4.0.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+inherit cmake
+
+DESCRIPTION="Simple Open (Database) Call Interface - write SQL in C++"
+HOMEPAGE="https://github.com/SOCI/soci"
+SRC_URI="https://github.com/${PN}/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+IUSE="boost firebird mysql odbc oracle postgres +sqlite test"
+RESTRICT="!test? ( test )"
```
Comment (AndrewAmmerlaan) [inline dev-libs/soci/soci-4.0.1.ebuild:12]: Try to stick to the ebuild skeleton as much as possible, i.e. IUSE and RESTRICT below KEYWORDS: https://wiki.gentoo.org/wiki/Basic_guide_to_write_Gentoo_Ebuilds

Comment (ezzieyguywuf) [reply dev-libs/soci/soci-4.0.1.ebuild]: fixed

File: dev-libs/soci/soci-4.0.1.ebuild (hunk 3 above)
Comment (ezzieyguywuf) [reply dev-libs/soci/soci-4.0.1.ebuild]: done.

File: dev-libs/soci/metadata.xml (hunk 2 above)
Comment (ezzieyguywuf) [reply dev-libs/soci/metadata.xml]: done

File: dev-libs/soci/metadata.xml (hunk 1 above)
Comment (ezzieyguywuf) [reply dev-libs/soci/metadata.xml]: done

File: dev-libs/soci/files/soci-4.0.3-cmake-config-path.patch [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ # Core configuration summary
+--- a/cmake/SociBackend.cmake
++++ b/cmake/SociBackend.cmake
+@@ -229,7 +229,8 @@ macro(soci_backend NAME)
+          )
+       endif()
+ 
+-      install(EXPORT SOCI NAMESPACE SOCI:: DESTINATION cmake)
++      install(EXPORT SOCI NAMESPACE SOCI:: DESTINATION 
++              "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/cmake")
```
Comment (SoapGentoo) [inline dev-libs/soci/files/soci-4.0.3-cmake-config-path.patch:24]:
```suggestion
+              "${CMAKE_INSTALL_LIBDIR}/cmake")
```

File: dev-libs/soci/files/soci-4.0.3-cmake-config-path.patch [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+--- a/src/core/CMakeLists.txt
++++ b/src/core/CMakeLists.txt
+@@ -124,7 +124,8 @@ if (SOCI_STATIC)
+     ARCHIVE DESTINATION ${LIBDIR})
+ endif()
+ 
+-install(EXPORT SOCI NAMESPACE SOCI:: DESTINATION cmake)
++install(EXPORT SOCI NAMESPACE SOCI:: DESTINATION
++    ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/cmake)
```
Comment (SoapGentoo) [inline dev-libs/soci/files/soci-4.0.3-cmake-config-path.patch:12]:
```suggestion
+    ${CMAKE_INSTALL_LIBDIR}/cmake)
```

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: dev-libs/soci/files/soci-4.0.3-cmake-config-path.patch (hunk 6 above)
Comment (ezzieyguywuf) [reply dev-libs/soci/files/soci-4.0.3-cmake-config-path.patch:12]: I thought i deleted this file.

Upstream is using GNUInstallDirs now, so the patch is no longer needed.

**Comment by thesamesam** [comment]
Not sure there's a point in doing this w/o doing it with rstudio and that work is stalled.

**Pull request closed without merging**